test/e2e-api/nats.go

```diff
@@ -33,12 +33,17 @@ func init() {
 
 	http.HandleFunc("/nats/pump-subject", func(w http.ResponseWriter, r *http.Request) {
 		subject := r.URL.Query().Get("subject")
-		mf := newMessageFactory(r.URL.Query())
```

Comment:
why was this replaced ?

Reply:
MessageFactory is implemented to generate messages using a prefix and n, so it's generating a list of data as prefix-1, prefix-2, ... prefix-n. With us switching from regex matching to exact matching, we are no longer able to verify for prefix-* regex. I remove this and instead always publish exact same data so that we can use exact match to verify.